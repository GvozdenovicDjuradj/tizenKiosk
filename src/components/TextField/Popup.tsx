import React from "react"
import {
  Animated,
  StyleProp,
  Text,
  ViewStyle,
} from "react-native"
import { popup as styles } from "./styles"

export const SPEED = 500

interface Props {
  show: boolean;
  style?: StyleProp<ViewStyle>;
  text: string;
}

export default class Popup extends React.PureComponent<Props> {

  private readonly _opacity: Animated.Value = new Animated.Value(0)

  public componentWillReceiveProps(nextProps: Props) {
    const { show } = this.props
    if (show !== nextProps.show) {
      Animated.timing(this._opacity, {
        duration: SPEED,
        toValue: Number(nextProps.show)
      }).start()
    }
  }

  public render() {
    const { style, text } = this.props
    const opacity = { opacity: this._opacity }
    return (
      <Animated.View pointerEvents="none" style={[opacity, styles.popup, style]}>
        <Text style={styles.popupText}>{text}</Text>
      </Animated.View>
    )
  }
}
