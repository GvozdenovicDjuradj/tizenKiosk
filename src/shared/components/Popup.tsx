import React, {useState, FC} from "react"
import styled from "styled-components/native"
import {fonts} from "../../theme/fonts"
import {Animated} from "react-native"

const Container = styled(Animated.View)`
    position: absolute;
    width: 105%;
    background-color: black;
    padding-left: 20px;
    padding-right: 20px;
    bottom: 50px;
    left: -1px;
`

const Information = styled.Text`
    font-family: ${fonts.get("ubuntu")};
    font-size: 20px;
    text-align: center;
    color: white;
`

export interface Props {
    message: string
    isOpen: boolean
}

const Popup: FC<Props> = ({ message, isOpen }) => {
    const [fadeAnim] = useState<Animated.Value>(new Animated.Value(isOpen ? 1 : 0))

    React.useEffect(() => {
        Animated.timing(
            fadeAnim,
            {
                toValue: isOpen ? 1 : 0,
                duration: 500,
            }
        ).start()
    }, [isOpen])

    return (
        <Container style={{ opacity: fadeAnim }} pointerEvents="none" >
            <Information>{message}</Information>
        </Container>
    )
}

export default Popup
