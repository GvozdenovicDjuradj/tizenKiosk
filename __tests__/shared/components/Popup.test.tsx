import React from "react"
import renderer from "react-test-renderer"
import Popup from "../../../src/shared/components/Popup"

describe(`the <Popup /> component`, () => {
    it(`renders correctly`, () => {
        // When
        const tree = renderer.create(<Popup isOpen={true} message="Example Message..." />).toJSON()

        // Then
        expect(tree).toMatchSnapshot()
    })
})
