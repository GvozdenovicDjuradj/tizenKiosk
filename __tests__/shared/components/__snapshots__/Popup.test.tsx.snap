// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <Popup /> component renders correctly 1`] = `
<View
  pointerEvents="none"
  style={
    Object {
      "backgroundColor": "black",
      "bottom": 50,
      "left": -1,
      "opacity": 1,
      "paddingLeft": 20,
      "paddingRight": 20,
      "position": "absolute",
      "width": "105%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "Ubuntu",
          "fontSize": 20,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Example Message...
  </Text>
</View>
`;
